src/dataIntegration/transcriptomics/call_MBA.m

```diff
@@ -0,0 +1,232 @@
+%to ensure activity int the extracted model. Medium confidence reactions
+%are only kept when a certain parsimony trade-off is met. In random order,
+%the algorithm prunes other reactions and removes them if not reauired to
+%support high- or medium- confidence reactions.
+%
+%INPUTS
+%
+%   model               input model (COBRA model structure)
+%   expressionRxns      expression data, corresponding to model.rxns (see
+%                       mapGeneToRxn.m)
+%   threshold_medium    reactions with expression above this threshold are medium
+%                       confidence (expression threshold)
+%   threshold_high      reactions with expression above this threshold are high confidence
+%                       (expression threshold)
+%   tol                 tolerance by which reactions are defined inactive after model extraction
+%                       (recommended lowest value 1e-8 since solver
+%                       tolerance is 1e-9)
+%   core                cell with reaction names (strings) that are manually put in
+%                       the high confidence core
+%   epsil               trade-off between removing medium confidence and
+%                       low confidence reactions (usually 0.5)
+%
+%OUTPUTS
+%
+%   tissueModel         extracted model
+%
+%* Jerby et al. (201)). Computational reconstruction of tissue-specific
+%metabolic models: application to human liver metabolism. Mol. Syst. Biol.
+%6, 401.
+%
+
+    % Get High expression core and medium expression core
+    indH = find(expressionRxns > threshold_high);
+    indM = find(expressionRxns >= threshold_medium & expressionRxns <= threshold_high);
+    CH = union(model.rxns(indH),core);
+    CM = model.rxns(indM);
+    
+    NC = setdiff(model.rxns,union(CH,CM));
+    
+    %Biomass metabolite sinks do not have to be pruned
+    nonBmsInd=cellfun(@isempty,strfind(NC, 'BMS_'));
+    NC = NC(nonBmsInd);
+    
+    %MBA
+    PM = model;
+    removed = {};
+    while ~isempty(NC)
+        ri = randi([1,numel(NC)],1);
+        r = NC{ri};
+        inactive = CheckConsistency(PM, r, tol);%% TO DO - need to uniformize the use of this function among all the different methods
+        eH = intersect(inactive, CH);
+        eM = intersect(inactive, CM);
+        eX = setdiff(inactive,union(CH,CM));
+        if numel(eH)==0 && numel(eM) < epsil*numel(eX)
+            PM = removeRxns(PM, inactive);
+            NC = setdiff(NC,inactive);
+            removed = union(removed,inactive);
+        else
+            NC = setdiff(NC,r);
+        end
+    end
+
+
+    tissueModel = removeNonUsedGenes(PM);
+    
+    is_active = fastcc(tissueModel, tol);
+    inactiveRxns = setdiff(tissueModel.rxns, tissueModel.rxns(is_active));
+    if ~isempty(inactiveRxns)
+        warning('Extracted model is not consistent, this might be caused by (numerical) issues in fastcc consistency checks')
+    end
+end
+
+function inactiveRxns = CheckConsistency(model, r, epsilon)

```

Comment:
I reallly think we can end up with one unique method used to check the consistency of a extracted model. For the moment the different extraction methods either use findBlockedReaction, checkConsistencyModel (that uses fastCC) or directly fastCC. Personnaly, i would go for simplicity, by using a FVA check of the model and setting a tolerance threshold that can be modulated... this way the user really easily understand what the function stand for as the FVA check is a widely accepted consistency check